capstone/capdb/utils.py

```diff
@@ -1,36 +1,30 @@
 from django.template.defaultfilters import slugify
 
 
-def generate_unique_slug(model, field, raw_string, count=None):
+def generate_unique_slug(instance, raw_string, field='slug', max_tries=1000):
     """
-    :param field: the field requiring unique slug
-    :param raw_string: basis for slug
-    :return:
-    Rejection sampling: generate unique slug, check for uniqueness
-    if found in model, resample
+        Get a unique slug for instance by sluggifying raw_string, checking database, and appending -1, -2 etc. if necessary.
+        When checking uniqueness, ignore this object itself if it already exists in the database.
+
```

Comment:
Is doing the pycharm recommended :param param_name: thing a faux pas?

Reply:
My gut feeling is the :param: thing isn't necessary -- most params are self-explanatory (or can be with good names), we're not generating HTML documentation, and we're unlikely to document :params: consistently. I feel like the time would be better spent documenting how to call the function or adding other comments. But maybe this is just laziness on my part, could be talked into it.

Reply:
Nah, that makes total sense. I like the conversational aspect of this anyway — and, to be honest, simply switched from this kind of thing (though I wasn't great about writing docstrings to begin with) to the other because of PyCharm autofill.